src/PointwiseFunctions/Hydro/EquationsOfState/Spectral.cpp

```diff
@@ -0,0 +1,309 @@
+    return Scalar<double>{pressure_from_density(get(rest_mass_density))};
+  } else if constexpr (std::is_same_v<DataType, DataVector>) {
+    auto result = make_with_value<Scalar<DataVector>>(rest_mass_density, 0.0);
+    for (size_t i = 0; i < get(result).size(); ++i) {
+      get(result)[i] = pressure_from_density(get(rest_mass_density)[i]);
+    }
+    return result;
+  }
+}
+
+template <class DataType>
+Scalar<DataType> Spectral::rest_mass_density_from_enthalpy_impl(
+    const Scalar<DataType>& specific_enthalpy) const {
+  if constexpr (std::is_same_v<DataType, double>) {
+    return Scalar<double>{
+        rest_mass_density_from_enthalpy(get(specific_enthalpy))};
+  } else if constexpr (std::is_same_v<DataType, DataVector>) {
+    auto result = make_with_value<Scalar<DataVector>>(specific_enthalpy, 0.0);
+    for (size_t i = 0; i < get(result).size(); ++i) {
+      get(result)[i] =
+          rest_mass_density_from_enthalpy(get(specific_enthalpy)[i]);
+    }
+    return result;
+  }
+}
+
+template <class DataType>
+Scalar<DataType> Spectral::specific_enthalpy_from_density_impl(
+    const Scalar<DataType>& rest_mass_density) const {
+  if constexpr (std::is_same_v<DataType, double>) {
+    return Scalar<double>{
+        specific_enthalpy_from_density(get(rest_mass_density))};
+  } else if constexpr (std::is_same_v<DataType, DataVector>) {
+    auto result = make_with_value<Scalar<DataVector>>(rest_mass_density, 0.0);
+    for (size_t i = 0; i < get(result).size(); ++i) {
+      get(result)[i] =
+          specific_enthalpy_from_density(get(rest_mass_density)[i]);
+    }
+    return result;
+  }
+}
+
+template <class DataType>
+Scalar<DataType> Spectral::specific_internal_energy_from_density_impl(
+    const Scalar<DataType>& rest_mass_density) const {
+  if constexpr (std::is_same_v<DataType, double>) {
+    return Scalar<double>{
+        specific_internal_energy_from_density(get(rest_mass_density))};
+  } else if constexpr (std::is_same_v<DataType, DataVector>) {
+    auto result = make_with_value<Scalar<DataVector>>(rest_mass_density, 0.0);
+    for (size_t i = 0; i < get(result).size(); ++i) {
+      get(result)[i] =
+          specific_internal_energy_from_density(get(rest_mass_density)[i]);
+    }
+    return result;
+  }
+}
+
+template <class DataType>
+Scalar<DataType> Spectral::chi_from_density_impl(
+    const Scalar<DataType>& rest_mass_density) const {
+  if constexpr (std::is_same_v<DataType, double>) {
+    return Scalar<double>{chi_from_density(get(rest_mass_density))};
+  } else if constexpr (std::is_same_v<DataType, DataVector>) {
+    auto result = make_with_value<Scalar<DataVector>>(rest_mass_density, 0.0);
+    for (size_t i = 0; i < get(result).size(); ++i) {
+      get(result)[i] = chi_from_density(get(rest_mass_density)[i]);
+    }
+    return result;
+  }
+}
+
+template <class DataType>
+Scalar<DataType> Spectral::kappa_times_p_over_rho_squared_from_density_impl(
```

Comment:
Is there a reason this is unimplemented?

Reply:
It is zero for all "cold" equations of state, by definition; it is implemented, but trivial.

Reply:
Oh I think I understand, the partials in the definition are taken at constant rho?  Ok yeah this is fine as long as everything is as expected.